src/summary/history/allergy-edit-form.component.tsx

```diff
@@ -0,0 +1,335 @@
+import { match } from "react-router";
+import {
+  getPatientAllergyByPatientUuid,
+  getAllergyReaction,
+  updatePatientAllergy,
+  deletePatientAllergy
+} from "./allergy-intolerance.resource";
+import { useCurrentPatient } from "@openmrs/esm-api";
+import { createErrorHandler } from "@openmrs/esm-error-handling";
+import SummaryCard from "../cards/summary-card.component";
+import style from "./allergy-form.css";
+import dayjs, { Dayjs } from "dayjs";
+
+export function AllergyEditForm(props: AllergyEditFormProps) {
+  const [patientAllergyReaction, setPatientAllergyReaction] = React.useState(
+    []
+  );
+  const [patientAllergy, setPatientAllergy] = React.useState(null);
+  const [allergyReaction, setAllergyReaction] = React.useState();
+  const [enableButtons, setEnableButtons] = React.useState(true);
+  const formRef = React.useRef<HTMLFormElement>(null);
+  const [reactionSeverityUuid, SetReactionSeverityUuid] = React.useState(null);
+  const [allergyComment, setAllergyComment] = React.useState(null);
+  const [updatedDate, setUpdatedDate] = React.useState(null);
+
+  const [
+    isLoadingPatient,
+    patient,
+    patientUuid,
+    patientErr
+  ] = useCurrentPatient();
+
+  const setCheckedValue = uuid => {
+    let returnValue: boolean;
+    for (let i = 0; i < patientAllergy.reactions.length; i++) {
+      if (patientAllergy.reactions[i].reaction.uuid === uuid) {
+        returnValue = true;
+        break;
+      } else {
+        returnValue = false;
+      }
+    }
+    return returnValue;
+  };
+
+  const handleAllergenReactionChange = event => {
+    if (event.target.checked == true) {
+      setPatientAllergyReaction([
+        ...patientAllergyReaction,
+        { uuid: event.target.value }
+      ]);
+    } else {
+      patientAllergyReaction.splice(
+        patientAllergyReaction.findIndex(
+          reaction => reaction.uuid == event.target.value
+        ),
+        1
+      );
+    }
+  };
+
+  const handleSubmit = event => {
+    event.preventDefault();
+    const allergy: Allergy = {
+      allergenType: patientAllergy.allergen.allergenType,
+      codedAllergenUuid: patientAllergy.allergen.codedAllergen.uuid,
+      severityUuid: reactionSeverityUuid,
+      comment: allergyComment,
+      reactionsUuid: patientAllergyReaction
+    };
+    const abortController = new AbortController();
+    updatePatientAllergy(
+      allergy,
+      patientUuid,
+      props.match.params,
+      abortController
+    ).then(response => {
+      response.status == 200 && navigate();
+    }, createErrorHandler);
+    return () => abortController.abort();
+  };
+
+  const handleDeletePatientAllergy = () => {
+    const abortController = new AbortController();
+    deletePatientAllergy(patientUuid, props.match.params, abortController).then(
+      response => {
+        response.status == 204 && navigate();
+      }
+    );
+  };
+
+  const handleCancelChanges = () => {
+    formRef.current.reset();
+  };
+
+  function navigate() {
```

Comment:
can't you use `react-router` for this instead of `window.href`
```suggestion
import { useHistory } from "react-router-dom";
const history = useHistory();
  function navigate() {
     history.push(`/patient/${patientUuid}/chart/allergy`);
  }
```